marconi-core/src/Marconi/Core/Experiment/Indexer/SQLiteIndexer.hs

```diff
@@ -181,37 +192,76 @@ handleSQLErrors value =
 
 -- | Run a list of insert queries in one single transaction.
 runIndexQueriesStep
-  :: (MonadIO m)
-  => (MonadError IndexerError m)
-  => SQL.Connection
+  :: SQL.Connection
   -> [Timed (Point event) event]
   -> [SQLInsertPlan event]
-  -> m ()
+  -> IO ()
 runIndexQueriesStep _ _ [] = pure ()
-runIndexQueriesStep c events xs =
+runIndexQueriesStep c events plan =
   let runIndexQuery (SQLInsertPlan planExtractor planInsert) = do
         let rows = planExtractor =<< events
         case rows of
           [] -> pure ()
           [x] -> SQL.execute c planInsert x
           _nonEmpty -> SQL.executeMany c planInsert rows
-   in either throwError pure <=< liftIO $
-        handleSQLErrors (SQL.withTransaction c $ Async.mapConcurrently_ runIndexQuery xs)
+   in Async.mapConcurrently_ runIndexQuery plan
+
+-- | Run a list of insert queries in one single transaction.
+runIndexPlan
+  :: SQL.Connection
+  -> [Timed (Point event) event]
+  -> [[SQLInsertPlan event]]
+  -> IO ()
+runIndexPlan c = traverse_ . runIndexQueriesStep c
```

Comment:
do we need a `withTransaction` here?

Reply:
No we don't, we already have a `withTransaction` in the `runIndexQueries` call to `runIndexPlan` that also stores the sync points (and `runIndexPlan` is not used anywhere else).